app/main.js

```diff
@@ -1,20 +1,26 @@
 /* eslint-env node */
-
 const electron = require("electron")
-const app = electron.app
-const BrowserWindow = electron.BrowserWindow
+const {app, BrowserWindow, ipcMain, Menu, shell} = electron
 const isDev = require("electron-is-dev")
-
+const { URL } = require("url")
+const defaultMenu = require("electron-default-menu")
 const updater = require("./updater")
 const logger = require("./logger")
 
+const {authorizeUser, fetchAccessToken} = require("./userAuthentication")
+
 let mainWindow
+let deepLinkURLOnReady = null
 
 logger.init()
 
 function createWindow () {
   logger.info("creating app window")
 
+  app.setAsDefaultProtocolClient("x-github-classroom")
```

Comment:
Does this need to be associated with `createWindow`? This can be called either by `app.on('ready', ...)` or `app.on('activate', ...)`, and this feels like a separate concern to launching the window. Thoughts?

Reply:
Moved this to a separate call in the `app.on('ready', ...)` event before calling `createWindow`. Figured it fit better there than potentially calling it multiple times when `app.on('activate', ...)` is fired. Does that work?